src/Sarif.Multitool.FunctionalTests/TestData/UrisMustBeValid/InvalidFileLocationUri_Expected.sarif

```diff
@@ -0,0 +1,93 @@
+{
+  "$schema": "http://json.schemastore.org/sarif-2.0.0",
+  "version": "2.0.0",
+  "runs": [
+    {
+      "tool": {
+        "name": "Sarif.Multitool",
+        "fullName": "Sarif.Multitool 2.0.0.0",
+        "version": "2.0.0.0",
+        "semanticVersion": "2.0.0",
+        "sarifLoggerVersion": "2.0.0.0",
+        "language": "en-US",
+        "properties": {
+          "Comments": "Command line tool to perform transformations of input files to SARIF.",
+          "CompanyName": "Microsoft",
+          "ProductName": "Microsoft SARIF SDK"
+        }
+      },
+      "results": [
+        {
+          "ruleId": "SARIF1003",
+          "level": "error",
+          "message": {
+            "arguments": [
+              "/runs/0/results/0/analysisTarget/uri",
+              "fi&le:///c:/src/file.c"
+            ]
+          },
+          "ruleMessageId": "Default",
+          "locations": [
+            {
+              "physicalLocation": {
+                "fileLocation": {
+                  "uri": "file:///C:/Code/sarif-sdk/src/Sarif.Multitool.FunctionalTests/TestData/UrisMustBeValid/InvalidFileLocationUri.sarif"
+                },
+                "region": {
+                  "startLine": 14,
+                  "startColumn": 43
+                }
+              }
+            }
+          ]
+        }
+      ],
+      "resources": {
+        "rules": {
+          "SARIF1003": {
+            "id": "SARIF1003",
+            "name": {
+              "text": "UrisMustBeValid"
+            },
+            "shortDescription": {
+              "text": "Specify a valid URI reference for every URI-valued property."
+            },
+            "fullDescription": {
+              "text": "Specify a valid URI reference for every URI-valued property."
+            },
+            "messageStrings": {
+              "Default": "{0}: The string \"{1}\" is not a valid URI reference."
+            },
+            "helpUri": "http://docs.oasis-open.org/sarif/sarif/v2.0/csprd01/sarif-v2.0-csprd01.html"
+          }
+        }
+      },
+      "files": {
+        "file:///C:/Code/sarif-sdk/src/Sarif.Multitool.FunctionalTests/TestData/UrisMustBeValid/InvalidFileLocationUri.sarif": {
+          "mimeType": "application/sarif-json"
+        }
+      },
+      "invocations": [
+        {
+          "startTime": "2018-08-15T23:04:47.640Z",
+          "endTime": "2018-08-15T23:04:48.453Z",
+          "toolNotifications": [
+            {
+              "id": "MSG001.AnalyzingTarget",
+              "physicalLocation": {
+                "fileLocation": {
```

Comment:
>fileLocation [](start = 17, length = 12)

is this file manually or auto-generated?

Reply:
It is generated by the script `Update-Expected.ps1` in the Sarif.Multitool.FunctionalTests directory. I run that script by hand to update the "expected" files for each rule I implement. I check the output to ensure that the "expected" content has exactly the results I expect.

---
In reply to: [210453490](https://github.com/Microsoft/sarif-sdk/pull/1018#discussion_r210453490) [](ancestors = 210453490)